server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java

```diff
@@ -687,8 +687,10 @@ private void deduplicate(QueryPlan queryPlan, int fetchSize)
     if (queryPlan instanceof GroupByTimePlan) {
       GroupByTimePlan plan = (GroupByTimePlan) queryPlan;
       // the actual row number of group by query should be calculated from startTime, endTime and interval.
-      fetchSize = Math
-          .min((int) ((plan.getEndTime() - plan.getStartTime()) / plan.getInterval()), fetchSize);
+      long interval = (plan.getEndTime() - plan.getStartTime()) / plan.getInterval();
```

Comment:
What about the plan.getInterval() is zero?  does the code has been checked before?

Reply:
yes. if you set interval = 0, it will throw exception: "Interval must more than 0"

Reply:
got it